booking/src/main/java/org/dcsa/conformance/standards/booking/checks/BookingChecks.java

```diff
@@ -0,0 +1,672 @@
+      }
+      if (polNode == null || polNode.isEmpty()) {
+        issues.add("Port of Load/Place of Receipt values must be provided");
+      }
+      if(!"SD".equals(receiptTypeAtOrigin) && ielNode != null) {
+        issues.add("Container intermediate export stop-off location should not be provided");
+      }
+      return issues;
+    });
+
+
+  private static final JsonContentCheck REQUESTED_CHANGES_PRESENCE = JsonAttribute.customValidator(
+    "Requested changes must be present for the selected Booking Status ",
+    (body) -> {
+      var bookingStatus = body.path("bookingStatus").asText("");
+      var issues = new LinkedHashSet<String>();
+      if (PENDING_CHANGES_STATES.contains(BookingState.fromWireName(bookingStatus))) {
+        var requestedChanges = body.get("requestedChanges");
+        if (requestedChanges == null) {
+          issues.add("requestedChanges is missing in allowed booking states %s".formatted(PENDING_CHANGES_STATES));
+        }
+      }
+      return issues;
+    }
+  );
+
+  private static final JsonContentCheck REASON_PRESENCE = JsonAttribute.customValidator(
+    "Reason field must be present for the selected Booking Status",
+    (body) -> {
+      var bookingStatus = body.path("bookingStatus");
+      var amendedBookingStatus = body.path("amendedBookingStatus");
+      var issues = new LinkedHashSet<String>();
+      var status = amendedBookingStatus.isMissingNode() || amendedBookingStatus.isNull() ? bookingStatus : amendedBookingStatus;
+      if (REASON_STATES.contains(BookingState.fromWireName(status.asText()))) {
+        var reason = body.get("reason");
+        if (reason == null) {
+          issues.add("reason is missing in the Booking States %s".formatted(REASON_STATES));
+        }
+      }
+      return issues;
+    }
+  );
+
+  private static final JsonContentCheck CHECK_ABSENCE_OF_CONFIRMED_FIELDS = JsonAttribute.customValidator(
+    "check absence of confirmed fields in non confirmed booking states",
+    body -> {
+      var issues = new LinkedHashSet<String>();
+      var bookingStatus = body.path("bookingStatus").asText("");
+      if (NON_CONFIRMED_BOOKING_STATES.contains(BookingState.fromWireName(bookingStatus))) {
+        if (body.hasNonNull("termsAndConditions")) {
+          issues.add("termsAndConditions must not be present in %s".formatted(bookingStatus));
+        }
+        if (body.hasNonNull("carrierClauses")) {
+          issues.add("carrierClauses must not be present".formatted(bookingStatus));
+        }
+        if (body.hasNonNull("charges")) {
+          issues.add("charges must not be present".formatted(bookingStatus));
+        }
+        if (body.hasNonNull("shipmentCutOffTimes")) {
+          issues.add("shipmentCutOffTimes must not be present".formatted(bookingStatus));
+        }
+        if (body.hasNonNull("advanceManifestFilings")) {
+          issues.add("advanceManifestFilings must not be present".formatted(bookingStatus));
+        }
+      }
```

Comment:
`trasnportPlan` and `confirmedEquipments` should not be present in a non-confirmed booking

Reply:
done